sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -342,6 +342,7 @@ trait CheckAnalysis extends PredicateHelper {
 
           case InsertIntoTable(t, _, _, _, _)
             if !t.isInstanceOf[LeafNode] ||
```

Comment:
This PR is to backport the fix from the master branch to Spark 2.0 branch. I think your comment is valid. You can submit a PR for improving the code style. Thanks!